File: leetcode/100-999/168. Excel Sheet Column Title/solution_1.js
/**
 * @param {number} columnNumber
 * @return {string}
 */

var convertToTitle = function(columnNumber) 
{
    let title = ""

    while (columnNumber > 0)
    {
        --columnNumber;
        const remainder = columnNumber % 26
        const letter = String.fromCharCode('A'.charCodeAt(0) + remainder)

        title = letter + title

        columnNumber = Math.floor(columnNumber / 26)
    }

    return title
};

File: leetcode/100-999/168. Excel Sheet Column Title/solution_1.cpp
class Solution 
{
    public:
        string convertToTitle(int columnNumber) 
        {
            string title;

            while (columnNumber > 0)
            {
                --columnNumber;
                int remainder = columnNumber % 26;
                char letter = 'A' + remainder;
                
                title = letter + title;

                columnNumber /= 26;
            }
            
            return title;
        }
};